packages/forms/src/components/ErrorSummary.tsx

```diff
@@ -1,17 +1,95 @@
 import React from "react";
 import { useIntl } from "react-intl";
-import { useFormState } from "react-hook-form";
+import { FieldErrors, FieldValues, useFormState } from "react-hook-form";
+import { ErrorMessage } from "@hookform/error-message";
 
 import {
   Alert,
   ScrollToLink,
   ScrollLinkClickFunc,
   Link,
 } from "@gc-digital-talent/ui";
-import { errorMessages, getLocale } from "@gc-digital-talent/i18n";
+import { commonMessages, errorMessages } from "@gc-digital-talent/i18n";
 import { notEmpty } from "@gc-digital-talent/helpers";
 
 import type { FieldLabels } from "./BasicForm";
+import { flattenErrors } from "../utils";
+
+type FieldNameWithLabel = {
+  label: React.ReactNode;
+  name: string;
+  index?: number;
+};
+
+const numberRegex = /\d/g;
```

Comment:
This will only match a single digit.  I found the error block won't appear for errors in the 11th field array item.

Reply:
Fixed this as well as the weird issue with selecting even numbers only 😆 

[update error label logic for nested fields](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8301/commits/f2a9433d995a9237123c147b4a3a75900baaa24e)

Reply:
Beautiful!